driver-pulsar/src/main/java/io/nosqlbench/driver/pulsar/PulsarActivity.java

```diff
@@ -197,6 +197,6 @@ public void failOnAsyncOperationFailure() {
     }
 
     public void asyncOperationFailed(Throwable ex) {
-        this.asyncOperationFailure = asyncOperationFailure;
+        this.asyncOperationFailure = ex;
```

Comment:
before this fix `failOnAsyncOperationFailure()` could never throw...

Reply:
good catch !